### PR TITLE
[RFC] Network design support

### PR DESCRIPTION
**Pull request by imobachgs**
This document summarizes the approach we are considering to follow to provide proper network configuration support in Agama. It is still under discussion.

See https://trello.com/c/lOo6xFUH/

### CODE REVIEW

File: doc/network_design_support.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Considered options
+
+Based on the situation described above, we considered these approaches:
+
+1. Implement support to set up the network through Agama D-Bus service.
+2. Keep the status quo, extend the web UI and rely on `nmcli` for the CLI-based installation. For
+   automation, we could rely on third-party tool.
+
+Although it might be harder, option 1 looks more consistent: you just need Agama D-Bus interface to
```
Comment (dgdavid) [inline doc/network_design_support.md:55]: +1

File: doc/network_design_support.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+* It is Ruby-based, so we might consider a Rust-based solution.
+
+### The proposal
+
+Our proposal is to build a D-Bus service that wraps around a third-party tool that takes care of the
+hard part. We considered [Netplan](https://netplan.io/) and [nmstate](https://nmstate.io/), although
+we decided to use the latter (see [Third-party tools](#third-party-tools)).
+
+Unfortunately, those tools are missing support for some cases (e.g., wireless configuration or udev
+handling), but we can such a support in our wrapper.
```
Comment (dgdavid) [inline doc/network_design_support.md:80]: missing verb?

```suggestion
handling), but we can add such a support in our wrapper.
```

File: doc/network_design_support.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+good enough interface to support our use cases.
+
+### Why not YaST2?
+
+You might be wondering, why not use YaST2 itself? Let's see some reasons:
+
+* It does not implement support for reading the NetworkManager configuration.
+* It is not able to talk to NetworkManager D-Bus interface. It configures NetworkManager by writing
+  the connection files.
+* It is Ruby-based, so we might consider a Rust-based solution.
```
Comment (jreidinger) [inline doc/network_design_support.md:71]: I would mention here that issue is not language, but memory requirements, which is quite high in ruby + yast2

File: doc/network_design_support.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Our proposal is to build a D-Bus service that wraps around a third-party tool that takes care of the
+hard part. We considered [Netplan](https://netplan.io/) and [nmstate](https://nmstate.io/), although
+we decided to use the latter (see [Third-party tools](#third-party-tools)).
+
+Unfortunately, those tools are missing support for some cases (e.g., wireless configuration or udev
+handling), but we can add such a support in our wrapper.
+
+## Third-party tools
+
+Both [Netplan](https://netplan.io/) and [nmstate](https://nmstate.io/) are tools that allow
```
Comment (jreidinger) [inline doc/network_design_support.md:84]: `are tools that allow` what?

File: doc/network_design_support.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- Based on devices/interfaces (the concept of connection does not exist).
+- Designed to support multiple network providers, but at this point only NetworkManager is
+  supported.
+- Offered as a Rust library.
+- Bindings for several languages and a CLI.
+- Support for WiFi and VPNs is missing.
+
+### Netplan
+
+- Written in Python.
```
Comment (jreidinger) [inline doc/network_design_support.md:97]: this looks like quite big downside for me.

File: doc/network_design_support.md (hunk 4 above)
Comment (imobachgs) [reply doc/network_design_support.md:84]: 🤦🏻

**Review by jreidinger — APPROVED** [review]

**Pull request merged**